client/src/components/BlogPostForm.tsx

```diff
@@ -0,0 +1,194 @@
+import { AiOutlineLeft } from 'react-icons/ai';
+import { Link } from 'react-router-dom';
+import { FaLocationDot } from 'react-icons/fa6';
+import { Entry, Photo } from '../lib/types';
+import { FormEvent, useState } from 'react';
+import UnsplashGallery from './UnsplashGallery';
+
+type props = {
+  entry: Entry | undefined;
+};
+
+export default function BlogPostForm({ entry }: props) {
+  const [title, setTitle] = useState(entry?.title ?? '');
+  const [subtitle, setSubtitle] = useState(entry?.subtitle ?? '');
+  const [location, setLocation] = useState(entry?.location ?? '');
+  const [photoInfo, setPhotoInfo] = useState(entry?.photoInfo ?? undefined);
+  const [body, setBody] = useState(entry?.body);
+  const [error, setError] = useState<unknown>();
+  const [search, setSearch] = useState('');
+  const [photos, setPhotos] = useState<Array<Photo>>();
```

Comment:
`Array<Photo>` is OK, but I prefer `Photo[]`. It's shorter, easier to type, and matches how you use it in code.